src/core/annotation.js

```diff
@@ -1188,6 +1187,60 @@ class FreeTextAnnotation extends MarkupAnnotation {
     super(parameters);
 
     this.data.annotationType = AnnotationType.FREETEXT;
+    const dict = parameters.dict;
+    this.pdfManager = parameters.pdfManager;
+
+    this.data.textColor = createRgbColor(dict.getArray("TextColor"));
+    this.data.textStyle = dict.get("DS");
+    this.data.fontSize = dict.get("FontSize");
+    this.data.opacity = dict.get("CA");
+    this.data.defaultAppearance = dict.get("DA");
+  }
+
+  getOperatorList(evaluator, task, renderForms) {
+    if (renderForms || this.appearance) {
+      return super.getOperatorList(evaluator, task, renderForms);
+    }
+
+    const data = this.data;
+
+    const appearanceDict = this.appearance.dict;
+    const appearanceStream = new Stream(stringToBytes(data.defaultAppearance));
+    const resourcesPromise = this.loadResources(["Font"]);
+    const bbox = appearanceDict.getArray("BBox") || [0, 0, 1, 1];
+    const matrix = appearanceDict.getArray("Matrix") || [1, 0, 0, 1, 0, 0];
+    const transform = getTransformMatrix(data.rect, bbox, matrix);
+
+    return resourcesPromise.then(resources => {
+      const opList = new OperatorList();
+      opList.addOp(OPS.beginAnnotation, [data.rect, transform, matrix]);
+      return evaluator
+        .getOperatorList({
+          stream: appearanceStream,
+          task,
+          resources,
+          operatorList: opList,
+        })
+        .then(() => {
+          const args = opList.argsArray;
+          for (let i = 0, ii = opList.fnArray.length; i < ii; i++) {
```

Comment:
`OperatorList.length` appears to [return `this.argsArray.length`](https://github.com/mozilla/pdf.js/blob/master/src/core/operator_list.js#L629-L632) instead of `this.fnArray.length`. Are they always the same length?